src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -228,6 +173,32 @@ public new CSharpParseOptions WithFeatures(IEnumerable<KeyValuePair<string, stri
             }
         }
 
+        internal override void ValidateOptions(ArrayBuilder<Diagnostic> builder)
+        {
+            ValidateOptions(builder, MessageProvider.Instance);
+
+            // Validate LanguageVersion not SpecifiedLanguageVersion, after Latest/Default has been converted:
+            if (!LanguageVersion.IsValid())
+            {
+                builder.Add(Diagnostic.Create(MessageProvider.Instance, (int)ErrorCode.ERR_BadLanguageVersion, LanguageVersion.ToString()));
+            }
+            
+            if (!PreprocessorSymbols.IsDefaultOrEmpty)
+            {
+                foreach (var symbol in PreprocessorSymbols)
+                {
+                    if (symbol == null)
+                    {
+                        builder.Add(Diagnostic.Create(MessageProvider.Instance, (int)ErrorCode.WRN_DefineIdentifierRequired, "null"));
```

Comment:
>WRN_DefineIdentifierRequired [](start = 95, length = 28)

Why are we reporting warnings here? If condition used to cause an exception, we should report an error instead, I believe.